src/PanelHeading.js

```diff
@@ -0,0 +1,52 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import cn from 'classnames';
+import elementType from 'react-prop-types/lib/elementType';
+
+import { prefix, bsClass, splitBsProps } from './utils/bootstrapUtils';
+
+const propTypes = {
+  componentClass: elementType,
+};
+
+const defaultProps = {
+  componentClass: 'div',
+};
+
+const contextTypes = {
+  $bs_panel: PropTypes.shape({
+    headingId: PropTypes.string,
+    bsClass: PropTypes.string
+  })
+};
+
+class PanelHeading extends React.Component {
+
+  render() {
+    const { children, className, componentClass: Component, ...props } = this.props;
+    const { headingId, bsClass: _bsClass } = this.context.$bs_panel || {};
+
+    const [bsProps, elementProps] = splitBsProps(props);
+    bsProps.bsClass = _bsClass || bsProps.bsClass;
+
+    if (headingId) {
+      elementProps.role = elementProps.role || 'tab';
+      elementProps.id = headingId;
+    }
+
+    return (
+      <Component
+        {...elementProps}
+        className={cn(className, prefix(bsProps, 'heading'))}
+      >
+        {children}
+      </Component>
+    );
+  }
+}
+
+PanelHeading.propTypes = propTypes;
+PanelHeading.defaultProps = defaultProps;
+PanelHeading.contextTypes = contextTypes;
+
+export default bsClass('panel', PanelHeading);
```

Comment:
lets defer that, we should do do big rethink of this api anyway, and reach some consistent policy for child components